platform/linux-generic/odp_classification.c

```diff
@@ -180,9 +180,17 @@ int odp_cls_capability(odp_cls_capability_t *capability)
 	capability->threshold_red.all_bits = 0;
 	capability->threshold_bp.all_bits = 0;
 	capability->max_hash_queues = CLS_COS_QUEUE_MAX;
+	capability->max_mark = ODP_CLS_MARK_DEFAULT;
 	return 0;
 }
 
+void odp_cls_pmr_create_opt_init(odp_pmr_create_opt_t *opt)
+{
+	opt->terms = NULL;
+	opt->num_terms = 0;
+	opt->mark = ODP_CLS_MARK_DEFAULT;
```

Comment:
This would actually be an illegal value with the proposed definition of max_mark. But see my comment about changing max_mark to actually mean max mark value.

Reply:
The default value will be 0 right? which is the one we need to set in odp_cls_pmr_create_opt_init(), Not sure what is illegal here? Please comment.

Reply:
If max_mark capability gets fixed so that it defines the maximum value that is still allowed, then this is ok. But if max_mark definition was not fixed, then it would require that the mark value in the rule parameters was at most _one less_ than zero (the default). Thus in that case this init function would set the mark value to something that one would not be allowed to pass to PMR creation.

So this comment was just to show that the current definition of max_mark in the capability struct just causes problems. But when that gets improved, this init function is ok.

Reply:
If I understand it correctly init will be filled with the default value(which is valid for all cases). So that application will first doodp_cls_pmr_create_opt_init() followed by if needs to change the mark value.

If we don't set the value as 0 then it can be any random value. Right? We need to avoid that. Right?

Reply:
Yes, all the fields need to be set by init() and zeroing the mark field is the right thing to do.

I meant that the code in this version of this PR sets max_mark capability to zero at the same time when the API says: "odp_pmr_create_opt_t::mark value must be less than this value.". Since zero is not less than zero, it is not ok to pass the options initialized by the init function to odp_cls_pmr_create() (I suppose you would not be able to call the function at all).

But this gets fixed when you fix the definition of max_mark so that it is the maximum mark value that is allowed. Then zero becomes an allowed value (and the only one) in this minimal implementation.

Reply:
Ack. I rephrased to "maximum value of odp_pmr_create_opt_t::mark". So code and doc will be aligned in the next version.